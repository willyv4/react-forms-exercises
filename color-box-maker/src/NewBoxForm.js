import { useState } from "react";

const NewBoxForm = ({ addBoxs }) => {
  const INITIAL_STATE = { bgColor: "", w: "", h: "" };
  const [formData, setFormData] = useState(INITIAL_STATE);

  const handleSubmit = (evt) => {
    evt.preventDefault();
    addBoxs(formData);
    setFormData(INITIAL_STATE);
  };

  const handleChange = (evt) => {
    const { name, value } = evt.target;
    setFormData((fData) => ({
      ...fData,
      [name]: value,
    }));
  };

  return (
    <form onSubmit={handleSubmit}>
      <label htmlFor="bgColor">Background Color:</label>
      <input
        id="bgColor"
        name="bgColor"
        value={formData.bgColor}
        onChange={handleChange}
      />

      <label htmlFor="w">Width:</label>
      <input id="w" name="w" value={formData.w} onChange={handleChange} />

      <label htmlFor="h">Height:</label>
      <input id="h" name="h" value={formData.h} onChange={handleChange} />
      <button>Add New Box!</button>
    </form>
  );
};

export default NewBoxForm;
