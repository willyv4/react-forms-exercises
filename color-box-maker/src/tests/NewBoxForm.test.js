import NewBoxForm from "../NewBoxForm";
import { render, fireEvent, screen } from "@testing-library/react";

it("renders without crashing", function () {
  render(<NewBoxForm />);
});

it("matches snapshot", function () {
  const { asFragment } = render(<NewBoxForm />);
  expect(asFragment()).toMatchSnapshot();
});

test("NewBoxForm submits new box data", () => {
  const addBoxs = jest.fn();

  render(<NewBoxForm addBoxs={addBoxs} />);

  fireEvent.change(screen.getByLabelText("Background Color:"), {
    target: { value: "red" },
  });
  fireEvent.change(screen.getByLabelText("Width:"), {
    target: { value: "100px" },
  });
  fireEvent.change(screen.getByLabelText("Height:"), {
    target: { value: "200px" },
  });

  fireEvent.click(screen.getByText("Add New Box!"));

  expect(addBoxs).toHaveBeenCalledWith({
    bgColor: "red",
    w: "100px",
    h: "200px",
  });
});
