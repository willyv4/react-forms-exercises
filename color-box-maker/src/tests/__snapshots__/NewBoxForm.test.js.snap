// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="bgColor"
    >
      Background Color:
    </label>
    <input
      id="bgColor"
      name="bgColor"
      value=""
    />
    <label
      for="w"
    >
      Width:
    </label>
    <input
      id="w"
      name="w"
      value=""
    />
    <label
      for="h"
    >
      Height:
    </label>
    <input
      id="h"
      name="h"
      value=""
    />
    <button>
      Add New Box!
    </button>
  </form>
</DocumentFragment>
`;
